### PR TITLE
[lit-html] Some fixes for styleMap()

### PR DESCRIPTION
**Pull request by justinfagnani**
* Accepts `undefined` and `null` property values, and removes the property. See #1616 for a PR on master
* Doesn't render `name: undefined;` for `{name: undefined}` on first render
* Makes `previousStyleProperties` re-namable to save some bytes

### CODE REVIEW

File: packages/lit-html/src/directives/style-map.ts [hunk 1]
```diff
@@ -64,37 +64,41 @@ class StyleMap extends Directive {
...
     // iteration.
-    this.previousStyleProperties!.forEach((name) => {
-      if (!(name in styleInfo)) {
-        this.previousStyleProperties!.delete(name);
-        if (name.indexOf('-') === -1) {
+    this._previousStyleProperties!.forEach((name) => {
+      // If the name isn't in styleInfo or it's null/undefined
+      if (styleInfo[name] == null) {
+        this._previousStyleProperties!.delete(name);
+        if (name.includes('-')) {
```
Comment (sorvell) [inline packages/lit-html/src/directives/style-map.ts:82]: Are we sure we're covered for this in IE11?

File: packages/lit-html/src/test/directives/style-map_test.ts [hunk 2]
```diff
@@ -77,13 +85,19 @@ suite('styleMap', () => {
...
+      marginTop: '2px',
+      'padding-bottom': '4px',
+      borderRadius: '5px',
+    });
     const el = container.firstElementChild as HTMLElement;
     assert.equal(el.style.marginTop, '2px');
     assert.equal(el.style.paddingBottom, '4px');
-    renderStyleMap({});
+    assert.equal(el.style.borderRadius, '5px');
+    renderStyleMap({borderRadius: undefined});
```
Comment (sorvell) [inline packages/lit-html/src/test/directives/style-map_test.ts:97]: Add an explicit `null`?

**Review by sorvell — APPROVED** [review]
LGTM with a couple nits

**Review by sorvell — CHANGES_REQUESTED** [review]
Oh, need to add this change to the changelog.

File: packages/lit-html/src/directives/style-map.ts (hunk 1 above)
Comment (justinfagnani) [reply packages/lit-html/src/directives/style-map.ts:82]: It's included in the core-js polyfills.

**Review by sorvell — APPROVED** [review]
LGTM

**Pull request merged**